transport/src/main/java/io/netty/channel/embedded/EmbeddedChannel.java

```diff
@@ -246,6 +248,20 @@ protected void initChannel(Channel ch) throws Exception {
         }
     }
 
+    private void setup(boolean register, final ChannelHandler handler) {
+        ChannelPipeline p = pipeline();
+        p.addLast(new ChannelInitializer<Channel>() {
```

Comment:
With the `EmbeddedChannel` the initializer always get triggered off `handlerAdded`, so I think we can get away with just directly adding the one handler to the pipeline, and forego the `ChannelInitializer`.

Reply:
Basically, the actual `handlerAdded` call is delayed (`AbstractChannel.register0`) until the completion of the `register` promise, so the channel is always registered when `handlerAdded` is called on the handlers added to the pipeline in in the `EmbeddedChannel.setup` methods.

Reply:
Was thinking about it, but wasn't sure. Let's try it. Thanks.